pkg/cmd/controller/csr.go

```diff
@@ -0,0 +1,44 @@
+package controller
+
+import (
+	"context"
+	"github.com/openshift/cluster-policy-controller/pkg/operator"
+	"github.com/openshift/library-go/pkg/operator/csr"
+	"github.com/openshift/library-go/pkg/operator/events"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	controllerName = "csr-approver-controller"
+	// Hardcoded for the Monitoring Operator:
+	// https://github.com/openshift/cluster-monitoring-operator/blob/master/manifests/0000_50_cluster-monitoring-operator_03-role-binding.yaml#L25-L27
+	monitoringServiceAccountNamespace = "openshift-monitoring"
+	monitoringServiceAccountName = "cluster-monitoring-operator"
+	monitoringCertificateSubject = "/CN=monitoring"
+)
+
+func RunCSRApproverController(ctx *ControllerContext) (bool, error) {
+	kubeClient, err := ctx.ClientBuilder.Client(infraClusterCSRApproverControllerServiceAccountName)
+	if err != nil {
+		return true, err
+	}
+
+	operatorClient := &operator.OperatorClient{
+	}
+
+	eventRecorder := events.NewRecorder(kubeClient.CoreV1().Events("test"), infraClusterCSRApproverControllerServiceAccountName, &v1.ObjectReference{})
+
+	controller := csr.NewCSRApproverController(
+		controllerName,
+		operatorClient,
+		kubeClient.CertificatesV1().CertificateSigningRequests(),
+		ctx.KubernetesInformers.Certificates().V1().CertificateSigningRequests(),
+		csr.NewLabelFilter(labels.Everything()),
+		csr.NewServiceAccountApprover(monitoringServiceAccountNamespace, monitoringServiceAccountName, monitoringCertificateSubject),
```

Comment:
@stlaz @s-urbaniak Both the CSR Filter and the Service Account Approver seem to do the same thing (in terms of functionality) - they filter somehow what CSRs are being approved. Perhaps we should create a Service Account Filter for them and squash both things into a single interface?

Reply:
They each serve a different purpose, I would be against such a unification

Reply:
agreeing with @stlaz, a service account approver is just a very specific filter validating username, groups, and subject.

Reply:
Ok, understood. Thanks for the context!